packages/recommend-react/src/__tests__/useRecommendations.test.ts

```diff
@@ -41,9 +42,108 @@ describe('useRecommendations', () => {
         },
         transformItems: (items) => items,
-      });
+      })
+    );
 
-      expect(recommendations).toEqual([hit]);
+    await waitFor(() => {
+      expect(result.current.recommendations).toEqual([hit]);
     });
   });
+
+  test('gets recommendations from initialState', async () => {
+    const { recommendClient } = createMockedRecommendClient();
+
+    const { result } = renderHook(
+      () =>
+        useRecommendations({
+          model: 'bought-together',
+          indexName: 'test',
+          recommendClient,
+          threshold: 0,
+          objectIDs: ['testing'],
+          queryParameters: {
+            facetFilters: ['test'],
+          },
+          fallbackParameters: {
+            facetFilters: ['test2'],
+          },
+          transformItems: (items) => items,
+          initialState,
+        }),
+      {
+        wrapper: React.StrictMode,
+      }
+    );
+
+    await waitFor(() => {
+      expect(result.current.recommendations).toEqual(
+        initialState.recommendations
+      );
+    });
+  });
+
+  test("doesn't initially fetch the recommendations with initialState", () => {
+    const { recommendClient } = createMockedRecommendClient();
+
+    renderHook(
+      () =>
+        useRecommendations({
+          model: 'bought-together',
+          indexName: 'test',
+          recommendClient,
+          threshold: 0,
+          objectIDs: ['testing'],
+          queryParameters: {
+            facetFilters: ['test'],
+          },
+          fallbackParameters: {
+            facetFilters: ['test2'],
+          },
+          transformItems: (items) => items,
+          initialState,
+        }),
+      {
+        wrapper: React.StrictMode,
+      }
+    );
+
+    expect(recommendClient.getRecommendations).toHaveBeenCalledTimes(0);
+  });
+
+  test('fetches recommendations when props change with initialState', async () => {
+    const { recommendClient } = createMockedRecommendClient();
+
+    const { rerender } = renderHook(
+      ({ indexName }) =>
+        useRecommendations({
+          model: 'bought-together',
+          indexName,
+          recommendClient,
+          threshold: 0,
+          objectIDs: ['testing'],
+          queryParameters: {
+            facetFilters: ['test'],
+          },
+          fallbackParameters: {
+            facetFilters: ['test2'],
+          },
+          transformItems: (items) => items,
+          initialState,
+        }),
+      {
+        wrapper: React.StrictMode,
+        initialProps: { indexName: 'test' },
+      }
+    );
+    expect(recommendClient.getRecommendations).toHaveBeenCalledTimes(0);
+
+    // we force a delay in order to simulate a prop change
+    // (that is a dependency prop in the useEffect from the hook we are testing)
+    // so that we can test that a network call is triggered
+    await forceDelay(1000);
+
+    rerender({ indexName: 'test1' });
+
+    expect(recommendClient.getRecommendations).toHaveBeenCalledTimes(1);
```

Comment:
We could assert that it was called with the correct `indexName`.

Reply:
✅ : https://github.com/algolia/recommend/pull/109/commits/466eafeb0672ced162687b66cc614c70d4b792f3